src/password.cpp

```diff
@@ -0,0 +1,45 @@
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <wordexp.h>
+
+#include "password.h"
+
+const std::string read_password(const char* file) {
+    wordexp_t exp;
+    std::ifstream ifs;
+    std::string pass = "";
+
+    wordexp(file,&exp,0);
+    if (exp.we_wordc > 0) {
+        ifs.open(exp.we_wordv[0]);
+    }
+    wordfree(&exp);
+    if (!ifs.is_open()) {
+    } else {
+        std::getline(ifs, pass);
+    }
+    return pass;
+}
+
+const std::string eval_password(const char* cmd) {
+    char buffer[128];
+    std::string pass = "";
+
+    FILE* pipe = popen(cmd, "r");
+    if (!pipe) {
+    } else {
+        try {
+            while (!feof(pipe)) {
+                if (fgets(buffer, 128, pipe) != NULL) {
+                    buffer[strcspn(buffer, "\r\n")] = 0;
```

Comment:
The intent was to cut on newlines, since most output is terminated by newlines. Also some tools like pass might return additional information (which are not the password) in the output, separated by newlines.
Since passwords can't contain newlines anyway (or at least I don't know any contrary), I thought this would be the most sensible.

If I recall correctly inserting a '\0' into the buffer (in this case replacing '\r' or '\n', whichever comes first, with '\0') should result in "terminating" the string and ignoring everything afterwards. At least this is the behavior I saw during testing.